test/rockpaperscissors.js

```diff
@@ -0,0 +1,405 @@
+
+contract('rcp', async accounts => {
+
+    const { toBN, soliditySha3, asciiToHex } = web3.utils;
+    const { getBalance } = web3.eth;
+    const playerOneSecretString = "secret1";
+    const playerTwoSecretString = "secret2";
+    const wrongSecretString = "wrongSecret";
+    
+    const ROCK = 0;
+    const PAPER = toBN(1);
+    const SCISSORS = toBN(2);
+
+    const INIT = "0";
+    const PLAYER_ONE_MOVE = "1";
+    const PLAYER_TWO_MOVE = "2";
+    const PLAYER_ONE_REVEAL = "3";
+
+    const INVALID_PLAYER_MSG = "Invalid player";
+    const GAME_NOT_FOUND_MSG = "Game not found";
+    const INVALID_MOVE_MSG = "Invalid move";
+    const INVALID_STEP_MSG = "Invalid step";
+    const HASH_MISMATCH_MSG = "Move and secret do not match";
+    const GAME_NOT_EXPIRED_MSG = "Game has not expired";
+    const MOVE_HASH_EMPTY_MSG = "Move hash is empty";
+
+    const noStakeToWithdrawMsg = "No stake to withdraw";
+    const badMatchMsg = "Move and secret do not match";
+    const alreadyMovedMsg = "Already moved";
+    const invalidPlayerMsg = "Invalid player";
+    const gameNotFoundMsg = "Game not found";
+    const gameInProgressMsg = "Game in progress";
+
+    const emptySecretMsg = "Secret cannot be empty";
+    const emptyHashErrorMsg = "Hash cannot be empty";
+    const noFundsAvailableMsg = "No funds available";
+
+    const ZERO_ADDRESS = "0x0000000000000000000000000000000000000000";
+    const ZERO_BYTES_32 = "0x0000000000000000000000000000000000000000000000000000000000000000";
+    const DAY_24_HOUR_IN_SECS = 86400;
+
+    const getBlockTimeStamp = async(txObj) => {
+
+        const tx = await web3.eth.getTransaction(txObj.tx);
+        const blockData = await web3.eth.getBlock(tx.blockNumber)
+
+        return blockData.timestamp
+    }
+
+    const checkEventNotEmitted = async () => {
+        const result = await truffleAssert.createTransactionResult(rcp, rcp.transactionHash);
+
+        await truffleAssert.eventNotEmitted(
+            result
+        );
+    };
+
+    const getGasCost = async txObj => {
+        const tx = await web3.eth.getTransaction(txObj.tx);
+
+        return toBN(txObj.receipt.gasUsed).mul(toBN(tx.gasPrice));
+    };
+
+    const [contractOwner, playerOne, playerTwo] = accounts;
+    const playerOneSecretBytes32 = await soliditySha3(playerOneSecretString);
+    const playerTwoSecretBytes32 = await soliditySha3(playerTwoSecretString);
+    const zeroPassBytes32 = await asciiToHex("");
+
+    const emptySha3Hash = await soliditySha3("");
+    const wrongSecretBytes32 = await soliditySha3(wrongSecretString);
+
+    let rcp;
+    let playerOneMoveHash;
+    let playerTwoMoveHash;
+    let gameKey;
+    let snapshotId;
+
+    beforeEach("Deploy and prepare", async function() {
+        rcp = await Game.new({from: contractOwner});
+        playerOneMoveHash = await rcp.createPlayerOneMoveHash(playerOneSecretBytes32, ROCK, {from: playerOne});
+        gameKey = playerOneMoveHash;
```

Comment:
Not sure why you would keep 2 variables then.

Reply:
I thought maybe to use a different name for different contexts as they are interchangeable. I will settle with gameKey.